File: source-code/kotlin/account-service/src/main/kotlin/com/vng/eventsourcing/Account/AccountService/Command/AccountCommand.kt
package com.vng.eventsourcing.Account.AccountService.Command

import com.vng.eventsourcing.Common.General.ICommand

/**
 * Created by Thuyen Phan
 * Package: com.vng.eventsourcing.Account.AccountService.Command
 * User: lap11852
 * Date: 27/01/2019
 * Time: 21:18
 */

class CreateAccount(val id : String, val username : String, val balance : Int) : ICommand()

class DepositMoney(val id : String, val amount : Int) : ICommand()

class WithdrawMoney(val id : String, val amount : Int) : ICommand()

File: src/pkg/service/account-service/account-command.go
package account

import (
	"github.com/billhcmus/cqrs/pkg/command"
)

// CreateAccount is user command
type CreateAccount struct {
	command.RootCommand
	Name string
}

// RechargeMoney is user command
type RechargeMoney struct {
	command.RootCommand
	Amount int64
}

// WithdrawMoney is user command
type WithdrawMoney struct {
	command.RootCommand
	Amount int64
}
